api/src/api.ts

```diff
@@ -154,5 +154,27 @@ export const plugin = {
         },
       },
     });
+
+    server.route({
+      method: "GET",
+      path: "papers/arxiv:{arxivId}/version",
+      handler: async (request, h) => {
+        const arxivId = request.params.arxivId;
+        const version = await dbConnection.getLatestProcessedArxivVersion({ arxiv_id: arxivId });
+        if (version < 0) {
+          // We don't have version info for this ID
+          return h.response().code(404);
+        }
+        // TODO (mjlangan): solidify the response structure (maybe 'maxVersion' or 'latestVersion'?)
+        return h.response({ version }).code(200);
+      },
+      options: {
+        validate: {
+          params: validation.arxivId.append({
+            arxivId: Joi.string().required(),
```

Comment:
Is this extra validation (`.required()`) necessary? I thought hapi would require arXiv ID to be specified by default due to how it parses the path, though I might be wrong.

Reply:
I'm not entirely sure....not that familiar with hapi myself. I might have been cargo-culting here 😅

Reply:
Hahahah nice! Then I think it should be good to take out. As far as I know, the parameters are already required by being part of the path spec above, and then these additional requirements do additional validation of the data in the params (such as length constraints, etc.)